processing/data_collection/gazette/spiders/rn_macaiba.py

```diff
@@ -0,0 +1,76 @@
+import re
+
+import dateparser
+import datetime as dt
+
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class RnMacaiba(BaseGazetteSpider):
+    start_date: dt.date
+    end_date: dt.date
+
+    TERRITORY_ID = "2407104"
+    EXTRA_ISSUE_PATTERN = r"Edição Extraordinária"
+    name = "rn_macaiba"
+    allowed_domains = ["macaiba.rn.gov.br"]
+
+    def __init__(self, start_date=None, end_date=None):
+        self.start_date = dt.date.today().replace(day=1)
+        self.end_date = dt.date.today()
+
+        if start_date is not None:
+            parsed_data = dateparser.parse(start_date)
+            if parsed_data is not None:
+                self.start_date = parsed_data.date()
+
+        if end_date is not None:
+            parsed_data = dateparser.parse(end_date)
+            if parsed_data is not None:
+                self.end_date = parsed_data.date()
+
+        self.logger.debug(
+            "Start date is {date}".format(date=self.start_date.isoformat())
+        )
+        self.logger.debug("End date is {date}".format(date=self.end_date.isoformat()))
+
+    def start_requests(self):
+        yield scrapy.FormRequest(
+            "https://www.macaiba.rn.gov.br/boletins/busca",
+            method="POST",
+            formdata={
+                "inicio": self.start_date.isoformat(),
+                "fim": self.end_date.isoformat(),
+            },
+        )
+
+    def parse(self, response):
+        for item in response.css("div.conteudo li.collection-item div"):
+            gazette = item.css("::text").get().strip()
+            date_match = re.findall(
+                r"(\d{2}) ([a-zç]+), (\d{4})", gazette, re.IGNORECASE
+            )
+            if len(date_match) != 1:
+                self.logger.error(f"Could not parse the date for Gazette {gazette}")
+                continue
+
+            item_date = " de ".join(date_match.pop())
+            item_date = dateparser.parse(item_date).date()
```

Comment:
I think a specific patter is better, relying on date parser can be a problem as it needs to identify the pattern and it can be a huge search through many defined patterns

Reply:
I've chosen to use this simply because it's already in use within the project.
Also, the month names within the page are in Portuguese, so I didn't want to mess with locales for date parsing.

Let me know your thoughts on this.

Reply:
That's fine, just constrain the language/locale `pt-br` then.
https://dateparser.readthedocs.io/en/latest/#dateparser.parse

Reply:
Done. I've restricted it to `pt` language, since the dateparser does not support `pt-BR`: https://github.com/scrapinghub/dateparser/issues/377